configs/platforms/windowsfips-2016-x64.rb

```diff
@@ -0,0 +1,28 @@
+platform "windowsfips-2016-x64" do |plat|
+  plat.vmpooler_template 'win-2016-fips-x86_64'
+  plat.servicetype 'windows'
+
+  # We need to ensure we install chocolatey prior to adding any nuget repos. Otherwise, everything will fall over
+  plat.add_build_repository "https://artifactory.delivery.puppetlabs.net/artifactory/generic/buildsources/windows/chocolatey/install-chocolatey-1.4.0.ps1"
+  plat.provision_with "C:/ProgramData/chocolatey/bin/choco.exe feature enable -n useFipsCompliantChecksums"
+
+  plat.add_build_repository "https://artifactory.delivery.puppetlabs.net/artifactory/api/nuget/nuget"
+
+  # We don't want to install any packages from the chocolatey repo by accident
+  plat.provision_with "C:/ProgramData/chocolatey/bin/choco.exe sources remove -name chocolatey"
+
+  #FIXME we need Fips Compliant Wix, currently not in choco repositories
+  #plat.provision_with "C:/ProgramData/chocolatey/bin/choco.exe install -y Wix310 -version 3.10.2 -debug -x86 --no-progress"
+  plat.provision_with "curl -L -o /tmp/wix314-binaries.zip https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/wix314-binaries.zip && 7z x -y -o\"/cygdrive/c/Program Files (x86)/WiX Toolset v3.14/bin\" /tmp/wix314-binaries.zip && rm /tmp/wix314-binaries.zip && SETX WIX \"C:\\Program Files (x86)\\WiX Toolset v3.14\" /M"
+  plat.provision_with "chmod -R 750 \"/cygdrive/c/Program Files (x86)/WiX Toolset v3.14/bin\""
```

Comment:
This is a result of using cygwin's 7z, which tries to emulate POSIX behavior of creating files with permissions 0644 (or whatever your emulated "umask" is). We then run cygwin's `chmod` command to make it executable, but because of the cywin/NTFS emulation, it's still not right. To restore permissions correctly, you have to run `icacls C:\Program Files (x86)\WiX Toolset v3.14 /T" and you may have to escape backslashes because the command is executed over ssh+bash.

Instead of that, I'd recommend installing `7z` from choco (see https://community.chocolatey.org/packages/7zip.install) and then using the full path to the binary when executing `C:\...\7z x -y -`

Reply:
Vanagon build: https://jenkins-platform.delivery.puppetlabs.net/view/vanagon-generic-builder/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/3255/

Artifact: https://builds.delivery.puppetlabs.net/puppet-agent/e28e127c8f3b42fb58fef0578982fd9104caae3d/artifacts/windowsfips/